src/common/containers/with-repository.js

```diff
@@ -0,0 +1,50 @@
+import React, { Component, PropTypes } from 'react';
+
+import { setGitHubRepository } from '../actions/repository-input';
+
+function withRepository(WrappedComponent) {
+
+  class WithRepository extends Component {
+
+    componentDidMount() {
+      if (!this.props.repository && this.props.fullName) {
+        this.props.dispatch(setGitHubRepository(this.props.fullName));
+      }
+    }
+
+    componentWillReceiveProps(nextProps) {
+      if (this.props.fullName !== nextProps.fullName) {
+        this.props.dispatch(setGitHubRepository(nextProps.fullName));
+      }
+    }
+
+    componentWillUnmount() {
+
+    }
+
+    render() {
+      const { fullName, ...passThroughProps } = this.props;
+      fullName; // just to 'use' this variable
```

Comment:
That's a lousy trick: I want to pass all the props except fullName to WrappedComponent, but I don't need fullName anywhere, so eslint complains about unused variable.

Or am I just tired and there is some obvious better way to do this?